server/routerlicious/packages/routerlicious-base/src/test/riddler/api.spec.ts

```diff
@@ -0,0 +1,115 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import * as crypto from "crypto";
+import express from "express";
+import request from "supertest";
+import { TestDbFactory } from "@fluidframework/server-test-utils";
+import { MongoManager, ISecretManager } from "@fluidframework/server-services-core";
+import * as riddlerApp from "../../riddler/app";
+
+const documentsCollectionName = "testDocuments";
+const deltasCollectionName = "testDeltas";
+const rawDeltasCollectionName = "testRawDeltas";
+
+class TestSecretManager implements ISecretManager {
+    constructor(
+        private readonly encryptionKey: string,
+    ) {
+    }
+
+    public decryptSecret(encryptedSecret: string): string {
+        return `test-decrypted-secret with key ${this.encryptionKey}`;
+    }
+
+    public encryptSecret(secret: string): string {
+        return `test-encrypted-secret with key ${this.encryptionKey}`;
+    }
+}
+
+describe("Routerlicious", () => {
+    describe("Riddler", () => {
+        describe("API", () => {
+            const document = {
+                _id: "doc-id",
+                content: "Hello, World!",
+            }
+            const defaultDbFactory = new TestDbFactory({
+                [documentsCollectionName]: [document],
+                [deltasCollectionName]: [],
+                [rawDeltasCollectionName]: [],
+            });
+            const defaultMongoManager = new MongoManager(defaultDbFactory);
+            const testTenantId = "test-tenant-id";
+            
+            let app: express.Application;
+            let supertest: request.SuperTest<request.Test>;
+
+            describe("CorrelationId", () => {
+                const correlationIdHeaderName = "x-correlation-id";
+                const testCorrelationId = "test-correlation-id";
+
+                const assertCorrelationId = async (url: string, method: "get" | "post" | "put" | "patch" | "delete" = "get"): Promise<void> => {
+                    await supertest[method](url)
+                        .set(correlationIdHeaderName, testCorrelationId)
+                        .then((res) => {
+                            assert.strictEqual(res.header?.[correlationIdHeaderName], testCorrelationId);
```

Comment:
(nit) `res.header?.[correlationIdHeaderName]` seems like incorrect syntax. Would expect it to be `res.header?[correlationIdHeaderName]`. It appears to compile and work just fine though 🤷‍♂️

Reply:
Not sure if this is Typescript specific, but if we do `res.header?[correlationIdHeaderName]` it will ask for another value to complete `res.header ? [correlationIdHeaderName] : somethingElse` expression.